ulauncher/search/apps/AppDb.py

```diff
@@ -155,28 +155,18 @@ def find(self, query, result_list=None):
         return result_list
 
 
-def search_name(name, exec_name):
+def get_exec_name(exec):
     """
-    Returns string that will be used for search
-    We want to make sure app can be searchable by its exec_name
+    Returns name of the executable file without the full path, env keyword and environment vars
+    Similar to Gio.DesktopAppInfo.get_executable(), except that will not drop env keyword and vars
     """
-    if not exec_name:
-        return name
-
-    # drop env vars
-    exec_name = ' '.join([p for p in exec_name.split(' ') if p != 'env' and '=' not in p])
-
-    # drop "/usr/bin/"
-    match = re.match(r'^(\/.+\/)?([-\w\.]+)([^\w\/]|$)', exec_name.lower(), re.I)
-    if not match:
-        return name
-
-    exec_name = match.group(2)
-    exec_name_split = set(exec_name.split('-'))
-    name_split = set(name.lower().split(' '))
-    common_words = exec_name_split & name_split
+    match = re.match(r'^(env\s+)?([^\s=]+=[^\s=]+\s+)*([^\s=]+\/)*(?P<bin>[^\s=]*)', exec, re.I)
```

Comment:
The env vars could be quoted values with spaces, right? 
```
>>> from ulauncher.search.apps.AppDb import get_exec_name
>>> get_exec_name('env MYVAR="value with spaces" /usr/bin/foo --option')
'with'
```

Reply:
though it's no worse than it was already and that seems to have worked well enough

Reply:
True. I didn't know how to support that without actually looping through character by character, because it could be single quotes as well, or single quotes with double quotes inside (or reversed). Possibly also backticks to run things through the shell. I think if we want to support that it'll add up to a lot more complex than a regex can support.

I ran `cat /usr/share/applications/*.desktop | grep Exec=` Ulauncher is the only app I have that even uses env.

Gio.DesktopAppInfo.get_executable() actually returns `env` instead of the app name for Ulauncher

Reply:
> Gio.DesktopAppInfo.get_executable() actually returns env instead of the app name for Ulauncher

ha! Yeah that is totally unhelpful for us. I suppose that is technically correct, since the `env` executable is in fact the executable that will be executed when the desktop file is run. It just happens that env will immediately turn around and exec a different executable based on its arguments! 

Regardless it does seem very uncommon to include any environment vars in desktop files, I also could not find any other examples

Reply:
It's also not absolutely critical that it gets the executable imo. It's just for apps which has known named like "Gedit", "Nautilus" etc that are named "Text Editor" and "Files" in their desktop entries. I thought about ditching this method entirely to go for Gio.DesktopAppInfo.get_executable(). If I wrote the initial implementation knowing all this, that's what I would do.